fix: Update Terraform GCP deployment instructions (#2099)
wandb/docs: platform/hosting/self-managed/gcp-tf.mdx

title: Deploy W&B Platform on Google Cloud
---

> <Note> W&B recommends fully managed deployment options such as [W&B Multi-tenant Cloud](/platform/hosting/hosting-options/multi_tenant_cloud) or [W&B Dedicated Cloud](/platform/hosting/hosting-options/dedicated_cloud/) deployment types. W&B fully managed services are simple and secure to use, with minimum to no configuration required.

Reviewer: Nit: Please put the opening tag on its own line, and clean up that extra whitespace before the W. For some reason, the Github UI isn't letting me make a suggestion.

   # Spin up all required services
   module "wandb" {
     source  = "wandb/wandb/google"

>      version = "~> 10.0"

Reviewer: Is there a plan to keep
this up to date?

Author: As much as possible. I took the opportunity to update that based on the latest version tested.
https://registry.terraform.io/modules/wandb/wandb/google/latest

To use a Google Cloud Storage bucket as a file storage backend for W&B, you will need to create a:

> - [PubSub Topic and Subscription](#create-pubsub-topic-and-subscription)

Reviewer: What changed here?

Author: As I mentioned in the PR description, Redis is no longer optional in the TF module.
The option to deploy Redis was used by the old Helm chart.

Reviewer: That's not what I was asking. This line showed up as a diff but I couldn't spot what changed. Maybe whitespace.

Author: There's no extra space or anything wrong, the content shifted, but the 3 topics aer altogether.
Please, check the dev branch.

### Create Storage Bucket

1. Navigate to the **Cloud Storage Buckets** page.
2. Select **Create bucket** and provide a name for your bucket. Ensure you choose a **Standard** [storage class](https://cloud.google.com/storage/docs/storage-classes).

Ensure that the service account or account that your instance is running has both:

> 

Reviewer: What changed here?

Author: Not sure if there's any change here. I think only the lines shifted because of the removal of the Redis block.

Reviewer: Same, I guess it is invisible whitespace.

Author: Not sure how to fix this. From my editor, I can't see anything.

- access to the bucket you created in the previous step
- `storage.objectAdmin` role on this bucket. For details, see https://cloud.google.com/storage/docs/access-control/using-iam-permissions#bucket-add

<Note>

>   Your instance also needs the `iam.serviceAccounts.signBlob` permission in

Reviewer: Please re-wrap this. We prefer a para to be on a single line.

- access to the bucket you created in the previous step
- `storage.objectAdmin` role on this bucket. For details, see https://cloud.google.com/storage/docs/access-control/using-iam-permissions#bucket-add

> <Note>Your instance also needs the `iam.serviceAccounts.signBlob` permission in Google Cloud to create signed file URLs. Add `Service Account Token Creator` role to the service account or IAM member that your instance is running as to enable permission.</Note>

Reviewer: Same feedback about the note tags, please put them on their own lines. :)

   - Content-Type
```

Note that the scheme, host, and port of the values for the origin must match exactly.

4. Make sure you have `gcloud` installed, and logged into the correct Google Cloud project.
5. Next, run the following:

```bash
gcloud storage buckets update gs://<BUCKET_NAME> --cors-file=<CORS_CONFIG_FILE>
```

### Create PubSub Notification

> 

Reviewer: Please remove

Author: Sorry, could you be more specific? What do you need to be removed?

Reviewer: Hmm, it looked like an extra blank line when I reviewed before, but it's fixed now.

Follow the procedure below in your command line to create a notification stream from the Storage Bucket to the Pub/Sub topic.

<Note>

>   You must use the CLI to create a notification stream. Ensure you have `gcloud`

Reviewer: Please re-wrap.

Follow the procedure below in your command line to create a notification stream from the Storage Bucket to the Pub/Sub topic.

> <Note> You must use the CLI to create a notification stream. Ensure you have `gcloud` installed.</Note>

Reviewer: Ditto

### Configure W&B server

1. Finally, navigate to the W&B `System Connections` page at `http(s)://YOUR-W&B-SERVER-HOST/console/settings/system`.
2. Select the provider `Google Cloud Storage (gcs)`,
3. Provide the name of the GCS bucket

<Frame>

>   <img

Reviewer: Please re-wrap.

    src="/images/hosting/configure_file_store_gcp.png"
    alt="Google Cloud file storage configuration"
  />
</Frame>

4. Press **Update settings** to apply the new settings.

## Upgrade W&B Server

> W&B is upgrade automatically by the W&B Operator if you wish to disable the automatic updates it's necessary to override the images using the system console and update the `UserSpec`.

Reviewer: ```suggestion
The Operator upgrades W&B automatically by the W&B Operator. To turn this off, you can update the user spec to override the images from the system console.
```

> Steps to pin the images to a certain version:

Reviewer: ```suggestion
To pin the images to a specific version:
```

> 1. Access the ActiveSpec in the system console `https://<W&B URL>/console/settings/advanced/spec/active`

Reviewer: ```suggestion
1. Access the active spec in the system console `https://<W&B URL>/console/settings/advanced/spec/active`
```

Author: I think we should keep the fidelity of the configuration name in the system console, in this case the configuration tab is `ActieSpec`

> 2. Copy all components image configuration. Ex.:

Reviewer: ```suggestion
2. Copy the components image configuration, which looks similar to:
```

> ```

Reviewer: ```suggestion
```yaml
```

  api:
    image:
      tag: 0.75.2
    initContainers:
      init-db:
        image:
          tag: 0.75.2
```

<Frame>

>   <img

Reviewer: Please re-wrap.

    src="/images/hosting/active_spec.png"
    alt="W&B System Console - ActiveSpec"
  />
</Frame>

> 3. Paste the component configuration replacing the image tag in the `UserSpec`

Reviewer: ```suggestion
3. Update the component configuration to replace the image tag version in the user spec:
```

Author: I think we should keep the fidelity of the configuration name in the system console, in this case the configuration tab is `UserSpec`

> ```

Reviewer: ```suggestion
```yaml
```

chart: {}
values:
  api:
    image:
      tag: 0.76.3
    initContainers:
      init-db:
        image:
          tag: 0.76.3
```

<Frame>

>   <img

Reviewer: Ditto

    src="/images/hosting/user_spec.png"
    alt="W&B System Console - UserSpec"
  />
</Frame>

> 4. Click Save

Reviewer: Add periods to list items. Also rewrap the image tag below.

5. Access the tab `Operator` and click `Trigger reapply`.
   <Frame>
     <img
       src="/images/hosting/operator.png"
       alt="W&B System Console - Operator"
     />
   </Frame>

> After these steps the W&B Operator will reconcile the configuration and pin the image version according to the configuartion override in the `UserSpec`

Reviewer: ```suggestion
The W&B Operator reconciles the configuration and pins the image to the specified version.
```

          tag: 0.76.3
```

<Frame><img src="/images/hosting/user_spec.png" alt="W&B System Console - UserSpec"/></Frame>

4. Click Save.

> 5. Access the tab `Operator` and click `Trigger reapply`.

Reviewer: I missed this before. Format UI labels with bold, not code font. :)
